ee/modules/500-operator-trivy/hooks/metrics_cis_benchmark.go

```diff
@@ -0,0 +1,136 @@
+			ApiVersion: "aquasecurity.github.io/v1alpha1",
+			Kind:       "ClusterComplianceReport",
+			LabelSelector: &v1.LabelSelector{
+				MatchLabels: map[string]string{
+					"module": "operator-trivy",
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"cis"},
+			},
+			FilterFunc: filterClusterComplianceReport,
+		},
+	},
+}, cisBencmarkMetricHandler)
+
+func filterClusterComplianceReport(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	compReport := &v1alpha1.ClusterComplianceReport{}
+
+	err := sdk.FromUnstructured(obj, compReport)
+	if err != nil {
+		return nil, err
+	}
+
+	filteredResult := filteredComplianceReport{}
+	switch {
+	case compReport.Status.DetailReport != nil:
+		filteredResult.DetailedControls = compReport.Status.DetailReport.Results
+	case compReport.Status.SummaryReport != nil:
+		filteredResult.SummaryControls = compReport.Status.SummaryReport.SummaryControls
+	}
+	return filteredResult, nil
+}
+
+func cisBencmarkMetricHandler(input *go_hook.HookInput) error {
+	input.MetricsCollector.Expire(metricGroupName)
+
+	snap := input.Snapshots[cisBenchmarkQueue]
+	if len(snap) == 0 {
+		input.LogEntry.Errorln("No CIS benchmark found")
+		return nil
+	}
+
+	compReport, ok := snap[0].(filteredComplianceReport)
+	if !ok {
+		return errors.New("can't use snapshot as filteredComplianceReport")
+	}
+
+	switch {
+	case compReport.DetailedControls != nil:
+		generateDetailedMetrics(input.MetricsCollector, compReport.DetailedControls)
+	case compReport.SummaryControls != nil:
+		generateSummaryMetrics(input.MetricsCollector, compReport.SummaryControls)
+	default:
+		input.LogEntry.Errorln("CIS benchmark didn't run")
+	}
+	return nil
+}
+
+func generateSummaryMetrics(metricsCollector go_hook.MetricsCollector, summaryChecks []v1alpha1.ControlCheckSummary) {
+	for _, controlCheck := range summaryChecks {
+		var (
+			totalFails float64
+		)
+
+		if controlCheck.TotalFail != nil && *controlCheck.TotalFail != 0 {
+			totalFails = float64(*controlCheck.TotalFail)
+		}
+
+		generateCisBenchmarkMetric(metricsCollector, totalFails, controlCheck.ID, controlCheck.Name, controlCheck.Severity)
+	}
+}
+
+func generateDetailedMetrics(metricsCollector go_hook.MetricsCollector, detailedChecks []*v1alpha1.ControlCheckResult) {
+	for _, controlCheck := range detailedChecks {
+		if controlCheck == nil {
+			continue
+		}
+		totalFails := countTotalFailsFromDetailedChecks(controlCheck.Checks)
+		generateCisBenchmarkMetric(metricsCollector, totalFails, controlCheck.ID, controlCheck.Name, controlCheck.Severity)
+	}
+}
+
+func countTotalFailsFromDetailedChecks(checks []v1alpha1.ComplianceCheck) float64 {
+	var totalFails float64
+	for _, check := range checks {
+		if !check.Success {
+			totalFails++
+		}
+	}
+	return totalFails
+}
+
+func generateCisBenchmarkMetric(metricsCollector go_hook.MetricsCollector, totalFails float64, id, name, severity string) {
+	metricsCollector.Set(
+		metricName,
+		totalFails,
```

Comment:
Should we remove metric if totalFails = 0 ?

Reply:
nope, because grafana dashboard is using all metrics to get percantage of passed checks (for example:
[`count(deckhouse_trivy_cis_benchmark{job=\"deckhouse\"} == 0)/count(deckhouse_trivy_cis_benchmark{job=\"deckhouse\"})`](https://github.com/deckhouse/deckhouse/pull/3995/files#diff-e9b6dad2d237ae459d6eb5ce0dc1c06b188a065af55dddeed6a9a78bcf75121aR123))